File: src/tools/definitions/web_search.py
from src.tools.types import Tool


def _execute(query: str) -> dict[str, list[str]]:
    # TODO: wire up real search API
    return {"results": [f"Placeholder result for: {query}"]}


web_search_tool = Tool(
    name="web_search",
    description="Search the web for information",
    parameters={
        "type": "object",
        "properties": {
            "query": {"type": "string", "description": "Search query"},
        },
        "required": ["query"],
    },
    execute=_execute,
)

File: src/tools/definitions/web-search.ts
import { z } from "zod";
import type { Tool } from "../types.js";

export const webSearchTool: Tool<
  { query: string },
  { results: string[] }
> = {
  name: "web_search",
  description: "Search the web for information",
  parameters: z.object({
    query: z.string().describe("Search query"),
  }),
  async execute({ query }) {
    // TODO: wire up real search API
    return { results: [`Placeholder result for: ${query}`] };
  },
};
